Harold/harold.py

```diff
@@ -34,62 +37,75 @@ def write(self, *args, **kwargs):
         print(*args, **kws)
         time.sleep(delay)
 
-    def __call__(self):
-        if not self.playing:
-            userlog = open("/home/pi/logs/user_log.csv", "a")
-            # Lower the volume during quiet hours... Don't piss off the RA!
-            self.mixer.setvolume(85 if quiet_hours() else 100)
-            varID = self.ser.readline()
-            print(varID)
-            # mplayer will play any files sent to the FIFO file.
-            if self.beep:
-                self.write("loadfile", DING_SONG)
-            if "ready" not in varID:
-                # Turn the LEDs off
-                LED.on(False)
-                # Get the username from the ibutton
-                uid, homedir = read_ibutton(varID)
-                # Print the user's name (Super handy for debugging...)
-                print("User: '" + uid + "'\n")
-                song = get_user_song(homedir, uid)
-                print("Now playing '" + song + "'...\n")
-                varID = varID[:-2]
-                userlog.write("\n" + time.strftime('%Y/%m/%d %H:%M:%S') + "," + uid + "," + song)
-                self.write("loadfile '" + song.replace("'", "\\'") + "'\nget_time_length",
-                           delay=0.0)
+    def play(self, override = False):
+        varID = self.ser.readline()
+        if varID == None or varID == "":
+            return
+        uid, homedir = read_ibutton(varID)
+        if uid in self.blacklist:
+            print("user %s is blacklisted, not playing" % uid)
+            return
+        if not (not override or (override and uid in self.override)) or uid == self.current_user:
```

Comment:
Don't worry, be happy! Stop being so negative.

(operator precedence with `not` and `or` might confuse people, and it doesn't read cleanly)